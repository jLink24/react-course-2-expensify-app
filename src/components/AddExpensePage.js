import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

// This component renders the "add expense page"
//      -Data from the expense form is passed up by the onSubmit function.  
//      -Data passed up is carried on an expense object argument
//      -After a sucessful submission, the onSubmit function then redirects the user to the homepage.  

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);  
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);