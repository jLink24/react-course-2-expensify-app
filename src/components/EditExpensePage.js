import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
    return (
        <div>
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense) => {
                    props.dispatch(editExpense(props.expense.id,expense));
                    props.history.push('/');
                }}
            />
            <button onClick={() => {
                props.dispatch(removeExpense({ id: props.expense.id }));
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
};

// This function returns an object that will be merged with the props object of the HOC
// The state is passed as the 1st argument and the props that will be merged into is the 2nd
const mapStateToProps = (state, props) => {
   return {
       expense: state.expenses.find((expense) => expense.id === props.match.params.id)
   } 
};

export default connect(mapStateToProps)(EditExpensePage);