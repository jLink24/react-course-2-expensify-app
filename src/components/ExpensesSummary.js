import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectedExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <p>Viewing {props.expenseCount} expenses totalling {numeral(props.expenseTotal / 100).format('$0,0.00')} </p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: selectedExpensesTotal(selectExpenses(state.expenses, state.filters))
    };
};

export default connect(mapStateToProps)(ExpensesSummary);