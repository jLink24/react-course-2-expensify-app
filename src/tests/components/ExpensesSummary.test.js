import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('should correctly render summary for 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render summary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={9434}/>);
    expect(wrapper).toMatchSnapshot();
});