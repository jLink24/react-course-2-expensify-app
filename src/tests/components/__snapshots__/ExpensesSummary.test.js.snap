// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary for 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expenses totalling 
    $94.34
     
  </p>
</div>
`;

exports[`should correctly render summary for 2 expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses totalling 
    $94.34
     
  </p>
</div>
`;
